lib/Models/SearchProviders/NominatimSearchProvider.ts

```diff
@@ -0,0 +1,140 @@
+import { makeObservable, runInAction } from "mobx";
+import Rectangle from "terriajs-cesium/Source/Core/Rectangle";
+import CesiumMath from "terriajs-cesium/Source/Core/Math";
+import { Feature, Point } from "@turf/helpers";
+import {
+  Category,
+  SearchAction
+} from "../../Core/AnalyticEvents/analyticEvents";
+import loadJson from "../../Core/loadJson";
+import LocationSearchProviderMixin from "../../ModelMixins/SearchProviders/LocationSearchProviderMixin";
+import NominatimSearchProviderTraits from "../../Traits/SearchProviders/NominatimSearchProviderTraits";
+import CreateModel from "../Definition/CreateModel";
+import Terria from "../Terria";
+import SearchProviderResults from "./SearchProviderResults";
+import SearchResult from "./SearchResult";
+
+export default class NominatimSearchProvider extends LocationSearchProviderMixin(
+  CreateModel(NominatimSearchProviderTraits)
+) {
+  static readonly type = "nominatim-search-provider";
+
+  get type() {
+    return NominatimSearchProvider.type;
+  }
+
+  constructor(uniqueId: string | undefined, terria: Terria) {
+    super(uniqueId, terria);
+
+    makeObservable(this);
+  }
+
+  protected logEvent(searchText: string) {
+    this.terria.analytics?.logEvent(
+      Category.search,
+      SearchAction.nominatim,
+      searchText
+    );
+  }
+
+  protected doSearch(
+    searchText: string,
+    searchResults: SearchProviderResults
+  ): Promise<void> {
+    searchResults.results.length = 0;
+    searchResults.message = undefined;
+
+    if (searchText === undefined || /^\s*$/.test(searchText)) {
```

Comment:
this check is unnecessary as searchText is always defined and is longer than min length. It is validated in SearchProviderMixin